### DISCUSSION TITLE
Using Jellyfin API key instead of username/password for authentication

### DISCUSSION THREAD

**Discussion by shahnanigans** [mueslimak3r/tv-intro-detection#23] [Ideas]
Topic  - how would I go about implementing this? I haven't delved into the codebase but this would make my docker setup much cleaner and easier to backup than hardcoding username/pass

**Comment by mueslimak3r** [comment]
This project uses jellyfin-apiclient-python.
https://github.com/jellyfin/jellyfin-apiclient-python

In this project, I think the implementation would be done by modifying `jellyfin_api_client.py`

It's likely that it supports creating a device session using an api key. There is an `AccessToken` which is saved when logging in but currently it isn't used in subsequent sessions.

Some examples for implementations might be found in these projects:
https://github.com/jellyfin/jellyfin-mpv-shim
https://github.com/jellyfin/jellyfin-kodi
